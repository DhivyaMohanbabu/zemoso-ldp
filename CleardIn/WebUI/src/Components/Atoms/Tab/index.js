import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Typography } from '@material-ui/core';
import PropTypes from 'prop-types';

const useStyles = makeStyles((theme) => ({
  tabSpan: {
    backgroundColor: '#eaefff',
    padding: '8px 12px 8px 20px',
    borderRadius: '5px 8px 0 0',
    color: theme.palette.clearedIn.dark,
    border: `1px solid ${theme.palette.clearedIn.grey[300]}`,
    cursor: 'pointer',
    textAlign: 'center',
    float: 'left',
    height: '36px',
    position: 'relative',
    boxSizing: 'border-box',
    maxWidth: 200,
    '&:after': {
      width: '16px',
      height: '36px',
      content: 'close-quote',
      backgroundColor: '#eaefff',
      position: 'absolute',
      top: '-1px',
      left: '100%',
      border: `1px solid ${theme.palette.clearedIn.grey[300]}`,
      borderLeft: 0,
      borderRadius: '0 5px 0 0',
      transform: 'skewX(30deg)',
      boxSizing: 'border-box',
      zIndex: 1,
    },
    '&:before': {
      width: '10px',
      height: '20px',
      content: 'close-quote',
      backgroundColor: '#eaefff',
      position: 'absolute',
      bottom: '-1px',
      left: '100%',
      borderRadius: ' 0 2px 0 0',
      borderBottom: `1px solid ${theme.palette.clearedIn.grey[300]}`,
      boxSizing: 'border-box',
    },
  },
  currentTab: {
    filter: 'drop-shadow(-1px -3px 3px rgba(50, 50, 0, 0.1))',
    backgroundColor: theme.palette.common.white,
    borderColor: theme.palette.common.white,
    zIndex: 2,
    color: theme.palette.clearedIn.blue,
    '&:after': {
      content: 'close-quote',
      backgroundColor: theme.palette.common.white,
      position: 'absolute',
      top: '0px',
      border: '0',
    },
    '&:before': {
      content: 'close-quote',
      backgroundColor: theme.palette.common.white,
      borderBottomColor: theme.palette.clearedIn.grey[300],
      position: 'absolute',
      bottom: '0px',
      border: '0',
    },
  },
}));

const Tab = ({ isSelected, title }) => {
  const classes = useStyles();

  return (
    <div className={`${classes.tabSpan} ${isSelected ? classes.currentTab : ''}`}>
      <Typography noWrap variant="body1">{title}</Typography>
    </div>
  );
};

Tab.propTypes = {
  isSelected: PropTypes.bool,
  title: PropTypes.string.isRequired,
};

Tab.defaultProps = {
  isSelected: false,
};

export default Tab;
