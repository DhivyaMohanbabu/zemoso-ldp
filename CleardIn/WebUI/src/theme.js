import { grey } from '@material-ui/core/colors';
import { createMuiTheme } from '@material-ui/core/styles';
// import 'typeface-roboto';

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#1665d8',
      light: '#b0f1f0',
    },
    secondary: {
      main: '#9ea0a5',
      dark: grey,
    },
    error: {
      main: '#ffada5',

    },
    success: {
      main: '#daefa9',
    },
    info: {
      main: grey[900],
      light: '#fdd570',
      dark: grey[600],
    },
    text: {
      primary: '#1c1c21', // '#3e3f42',
      secondary: '#5c5c5c', // #6b6c6f',
    },
    grey: {
      100: '#f8fafb',
      200: '#f2f4f6',
      300: '#e8ecef',
      400: '#98a9bc',
      500: '#778ca2',
    },
    clearedIn: {
      grey: {
        100: '#f6f6f6',
        200: '#eeeeee',
        300: '#cccccc',
        400: '#949494',
        500: '#666666',
      },
      selectedElementBGColor: '#eaf0ff', // rgba(77, 124, 254, 0.12),
      dark: '#5c5c5c',
      bodyBackground: '#fafafa',
      blue: '#1d4cd7',
    },
    relationShip: {
      vendor: '#daefa9',
      customer: '#fdd570',
      partner: '#b0f1f0',
    },
  },
  typography: {
    fontFamily: ['Roboto', 'Rubik', 'sans-serif'].join(','),
    h1: {
      fontSize: '1.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h2: {
      fontSize: '1.625rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h3: {
      fontSize: '1.25rem',
      fontWeight: 300,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h4: {
      fontSize: '1.125rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h5: {
      fontSize: '1rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    h6: {
      fontSize: '1.25rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 1.05,
      letterSpacing: 'normal',
    },
    subtitle1: {
      fontSize: '0.938rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    subtitle2: {
      fontSize: '0.938rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    caption: {
      fontSize: '0.875rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    link: {
      fontSize: '0.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    body1: {
      fontSize: '0.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.14,
      letterSpacing: 'normal',
    },
    body2: {
      fontFamily: 'Rubik',
      fontSize: '0.875rem',
      fontWeight: 300,
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    button: {
      fontSize: '0.75rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
      textTransform: 'none',
    },
  },
  overrides: {
    MuiButton: {
      root: {
        maxWidth: '200px',
        display: 'block',
      },
      label: {
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
      },
    },
  },
});

export default theme;
