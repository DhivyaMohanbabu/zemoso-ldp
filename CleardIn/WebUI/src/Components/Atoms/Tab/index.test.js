import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/core/styles';
import Tab from '.';
import theme from '../../../theme';

const allProps = {
  title: 'top 10',
};

it('renders inactive Tab correctly', () => {
  const { container, getByText } = render(
    <ThemeProvider theme={theme}>
      <Tab {...allProps} />
    </ThemeProvider>,
  );
  expect(container).toMatchSnapshot();
  expect(getByText(allProps.title)).toBeVisible();
});

it('renders active Tab correctly', () => {
  const { container, getByText } = render(
    <ThemeProvider theme={theme}>
      <Tab isSelected {...allProps} />
    </ThemeProvider>,
  );
  expect(container).toMatchSnapshot();
  expect(getByText(allProps.title)).toBeVisible();
});
