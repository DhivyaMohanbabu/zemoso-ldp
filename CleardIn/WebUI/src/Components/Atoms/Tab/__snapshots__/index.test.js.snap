// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders active Tab correctly 1`] = `
<div>
  <div
    class="makeStyles-tabSpan-3 makeStyles-currentTab-4"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap"
    >
      top 10
    </p>
  </div>
</div>
`;

exports[`renders inactive Tab correctly 1`] = `
<div>
  <div
    class="makeStyles-tabSpan-1 "
  >
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap"
    >
      top 10
    </p>
  </div>
</div>
`;
